### PR TITLE
Conditionally packaging a buildpack or an extension

### PR DESCRIPTION
**Pull request by pacostas**
## Summary

This PR changes the package.sh script to conditionally package a buildpack or an extension according to whether is an extension or a buildpack.

The way this is implemented is to check whether the extension file exists on the directory, otherwise, it fallbacks to the default behavior of packaging a buildpack.

Test results:
* node-engine: https://github.com/pacostas/node-engine/pull/48
* ubi-nodejs-extension: https://github.com/pacostas/ubi-nodejs-extension/pull/67

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: implementation/scripts/package.sh [hunk 1]
```diff
@@ -134,13 +140,28 @@ function buildpack::archive() {
...
 
-  util::print::title "Packaging buildpack..."
+  if [ "$buildpack_type" == "extension" ]; then
+    mkdir -p "${BUILD_DIR}/cnbdir"
 
-  pack \
-    buildpack package "${output}" \
-      --path "${BUILD_DIR}/buildpack.tgz" \
-      --format file
+    pushd "${BUILD_DIR}/cnbdir" > /dev/null
```
Comment (robdimsdale) [inline implementation/scripts/package.sh:150]: Initially i was confused why we were not doing something similar to the `pack buildpack package --path <path>` below, then I realized that `pack extension package` doesn't have that feature.

I opened [this feature request](https://github.com/buildpacks/pack/issues/2183) on pack -if/when that lands we can clean this up.

**Review by robdimsdale — DISMISSED** [review]
Looks good - thanks.

I cleaned up some code a bit, and left a comment inline for a potential future refactor.

**Comment by mhdawson** [comment]
@ForestEckhardt wondering if you might have time to help review/land?

File: implementation/scripts/package.sh [hunk 2]
```diff
@@ -64,20 +64,25 @@ function main {
 
   tools::install "${token}"
 
-  buildpack::archive "${version}"
-  buildpackage::create "${output}"
+  buildpack_type=buildpack
+  if [ -f "extension.toml" ]; then
```
Comment (ForestEckhardt) [inline implementation/scripts/package.sh:68]: Does this need to be an absolute path as well to solve the same problem as seen in https://github.com/paketo-buildpacks/github-config/pull/947

Comment (pacostas) [reply implementation/scripts/package.sh]: @ForestEckhardt There was no error on that one, but I changed it anyway to point to an absolute path. I also rebased the branch.

**Review by ForestEckhardt — APPROVED** [review]

**Pull request merged**